test/test_server.py

```diff
@@ -29,21 +29,17 @@ def test_local_port_forwarded(self):
 
     def test_local_port_scanning(self):
         with patch('uiautomator.next_local_port') as next_local_port:
-            self.Adb.return_value.forward_list.return_value = []
+            self.Adb.return_value.device_serial.return_value = "abcd"
+            self.Adb.return_value.forward_list.side_effect = [[], [["abcd", "tcp:1234", "tcp:9008"]]]
             next_local_port.return_value = 1234
             self.assertEqual(AutomatorServer("abcd", None).local_port,
                              next_local_port.return_value)
 
-            next_local_port.return_value = 14321
-            self.Adb.return_value.forward_list.return_value = Exception("error")
-            self.assertEqual(AutomatorServer("abcd", None).local_port,
-                             next_local_port.return_value)
-
     def test_device_port(self):
         self.assertEqual(AutomatorServer().device_port, 9008)
 
     def test_start_success(self):
-        server = AutomatorServer()
+        server = AutomatorServer(local_port=1234)
```

Comment:
Thoughts on using a helper to create the AutomatorServer so each test doesn't have to specify local_port?